stdlib/SuiteSparse/src/umfpack.jl

```diff
@@ -406,6 +406,32 @@ for itype in UmfpackIndexTypes
                         mx, mz, C_NULL, lu.numeric, umf_info)
             complex(mx[], mz[])
         end
+        function logabsdet(F::UmfpackLU{T, $itype}) where T # return log(abs(det)) and sign(det)
+            n = checksquare(F)
+            issuccess(F) || return log(zero(real(T))), log(one(T))
+            U = diag(F.U)
```

Comment:
I assume this is by far the most expensive step, right? Is there any way to get the diagonal entries directly? But we could leave that for later given that this is working correctly.

Reply:
This was originally indexing the diagonal directly as `U[i, i]`, which I changed, based on the (untested) hunch that indexing into a sparse matrix that way might be slower than extracting the diagonal all at once.  Either way, I just profiled the current version and it's not actually the most expensive part: most of the time is being spent in the inner loop, in lines 423 and 426.

Reply:
Hm, that's bad. Does it change anything if we go back to the `c += 1` version, but have one `c` and one `d` for `p` and `q`, respectively, to avoid overflow? Or we could "reduce" the value of `c`(and `d`) at the end of the outer loop like `c = ifelse(isodd(c), 1, 0)`. But I guess the main reason is the quadratic complexity, so we should find a better way to compute the sign from the permutation.

Reply:
Yes, I agree, I don't think incrementing `c` is going to be as big an issue as the quadratic complexity.